lib/system.nim

```diff
@@ -3043,3 +3043,6 @@ when defined(nimDebugUtils):
     {.define(nimCompilerDebug).}
     n
     {.undef(nimCompilerDebug).}
+
+proc `not`*(a: typedesc[ref], b: typeof(nil)): typedesc {.magic: "TypeTrait", noSideEffect.}
```

Comment:
```suggestion
proc `not`*[T: ref or ptr](a: typedesc[T], b: typeof(nil)): typedesc {.magic: "TypeTrait", noSideEffect.}
```
This should work. `liftParamType` doesn't recurse into `tyTypeDesc` (although it should), so my original `typedesc[ref or ptr]` won't work.

Reply:
Thanks!